network-area-diagram/src/main/java/com/powsybl/nad/svg/iidm/AbstractVoltageStyleProvider.java

```diff
@@ -108,4 +114,61 @@ protected Optional<String> getBaseVoltageStyle(BranchEdge edge, BranchEdge.Side
     }
 
     protected abstract Optional<String> getBaseVoltageStyle(Terminal terminal);
+
+    @Override
+    public List<String> getHighlightNodeStyleClasses(Node node) {
+        String subnetworkId = network.getIdentifiable(node.getEquipmentId()).getParentNetwork().getId();
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightSideEdgeStyleClasses(BranchEdge edge, BranchEdge.Side side) {
+        String subnetworkId = getSubnetworkId(edge, side);
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    @Override
+    public List<String> getHighlightThreeWtEdgStyleClasses(ThreeWtEdge edge) {
+        String subnetworkId = getSubnetworkId(edge.getEquipmentId(), edge.getSide());
+        return List.of(subnetworksHighlightMap.get(subnetworkId));
+    }
+
+    private String getSubnetworkId(BranchEdge edge, Side side) {
```

Comment:
you should take care of the case of returning null, the returned list should be empty then. Then it's easier that `getSubnetworkId` returns `Optional<String>` to handle this.